cmd/api-headers.go

```diff
@@ -157,9 +157,9 @@ func setObjectHeaders(w http.ResponseWriter, objInfo ObjectInfo, rs *HTTPRangeSp
 	}
 
 	if opts.PartNumber > 0 {
-		var start, end int64
+		var end = int64(-1)
 		for i := 0; i < len(objInfo.Parts) && i < opts.PartNumber; i++ {
-			start = end
+			start = end + 1
```

Comment:
A subtle difference would be where the loop never runs, than you will end up with `start = 0 and end = -1` in the RangeSpec, which I suspect should be `end = 0`.

It is a bit weird the way it is made since the byte at offset `end` is included.